src/NServiceBus.Core/Causation/ConversationIdStrategyContext.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace NServiceBus
+{
+    using Pipeline;
+
+    /// <summary>
+    /// Provides context when generating message conversation ID's.
+    /// </summary>
+    public class ConversationIdStrategyContext
```

Comment:
@bording went with a context class since I have a hunch we want to provide the message headers etc as well in the future. And this would allow that in a minor? (but yea YAGNI could be called)

Reply:
As it currently stands, I don't see value in this class, and it means we have another per-message heap allocation for it. If you think we need more than than just the message, I think I'd rather see that happen now.

Reply:
@bording if u care that much about allocations in the fast path then i suggest the @Particular/nservicebus-maintainers should reconsider this https://github.com/Particular/NServiceBus/pull/4872. if not then you should no longer use that as an argument for blocking change